cypress/integration/Modal.spec.ts

```diff
@@ -162,6 +162,132 @@ describe('Modal', () => {
+        beforeEach(() => {
+          cy.findByText('Open Popup to reach Delete button').click();
+        });
+
+        it(`should open the 'Really Delete' popup`, () => {
+          cy.findByRole('dialog', {name: 'Really Delete'}).should('be.visible');
+        });
+
+        context(`when the 'Delete' button is focused`, () => {
+          beforeEach(() => {
+            cy.findByRole('button', {name: 'Delete'}).focus();
+          });
+          it(`should open the 'Delete' tooltip`, () => {
+            cy.findByRole('tooltip', {name: 'Really, Really, Really, Really, Really sure'}).should(
+              'be.visible'
+            );
+          });
+
+          context(`when clicking outside the modal`, () => {
+            beforeEach(() => {
+              cy.get('body').click('top');
+            });
+
+            it(`should close the 'Delete' tooltip`, () => {
+              cy.findByRole('tooltip', {
+                name: 'Really, Really, Really, Really, Really sure',
+              }).should('not.be.visible');
+            });
+
+            it(`should close the 'Really Delete' popup`, () => {
+              cy.findByRole('dialog', {name: 'Really Delete'}).should('not.be.visible');
+            });
+
+            it(`should keep the modal open`, () => {
+              cy.findByRole('dialog', {name: 'Delete Item'}).should('be.visible');
+            });
+          });
+        });
+      });
+
+      context(`when the 'Non-hidable Popup' button is clicked`, () => {
+        beforeEach(() => {
+          cy.findByText('Non-hidable Popup').click();
+        });
+
+        it(`should open the 'Does Not Hide On Click Outside' popup`, () => {
+          cy.findByRole('dialog', {name: 'Does Not Hide On Click Outside'}).should('be.visible');
+        });
+
+        context(`when the 'Delete' button is focused`, () => {
+          beforeEach(() => {
+            cy.findByRole('button', {name: 'Delete'}).focus();
+          });
+          it(`should open the 'Delete' tooltip`, () => {
+            cy.findByRole('tooltip', {name: 'Really, Really, Really, Really, Really sure'}).should(
+              'be.visible'
+            );
+          });
+
+          context(`when clicking outside the modal`, () => {
+            beforeEach(() => {
+              // cy.findByRole('dialog', {name: 'Delete Item'}).click();
+              cy.get('body').click('top');
+            });
+
+            it(`should not close the 'Does Not Hide On Click Outside' popup`, () => {
+              cy.findByRole('dialog', {name: 'Does Not Hide On Click Outside'}).should(
+                'be.visible'
+              );
+            });
+
+            it(`should close the modal`, () => {
+              cy.findByRole('dialog', {name: 'Delete Item'}).should('not.be.visible');
+            });
+
+            // TODO: Manually, this test case works, but cypress click on body does not close the tooltip.
+            it.skip(`should close the 'Delete' tooltip`, () => {
```

Comment:
I could not get around this error and could not reproduce manually, but this is the error that appears in the DOM when the test runs: `Failed to execute 'removeChild' on 'Node'`

If someone has an idea of how to get around this error, that would be really helpful 🙏

Reply:
Update from offline conversation with Nicholas: this was a legitimate issue caught by cypress, and this will be fixed in v5, so for now, keeping this test in here and skipping it until we upgrade to v5.